bin/compile

```diff
@@ -21,4 +21,31 @@ source $HOME/.poetry/env
 echo "-----> Export requirements.txt from Poetry"
 
 cd "$BUILD_DIR"
+
+# pip can't handle editable dependencies when requiring hashes (https://github.com/pypa/pip/issues/4995)
 poetry export -f requirements.txt -o requirements.txt --without-hashes
+
+echo "-----> Export python runtime from Poetry"
+
+RUNTIME_FILE="runtime.txt"
+
+if [ -f "$RUNTIME_FILE" ]; then
+  echo "-----> $RUNTIME_FILE found, delete this file in your repository!"
+  exit 1
+fi
+
+# Get python version from poetry.lock file metadata section
+VERSION=$(sed -n '/^\[metadata\]/,/^\[/p' poetry.lock | sed -n -e 's/^python-versions\s*=\s*//p' | tr -d '\"')
+
+# Version only valid if exact version is specified
+# ^3.8 -> not valid
+# 3.8 -> not valid
+# 3.8.1 -> valid
+
+if [[ $VERSION =~ ^[0-9](\.[0-9]+){2}$ ]]; then
+  echo "-----> Write $VERSION into $RUNTIME_FILE"
+  echo "python-$VERSION" > $RUNTIME_FILE
+else
+  echo "-----> $VERSION not valid, please specify an exact python version in your pyproject.toml. (e.g. 3.8.1)"
```

Comment:
Genauer müßte man eigentlich sowas wie `pyproject.toml (and thus poetry.lock)` sagen.

Reply:
Updated suggestion.